### PR TITLE
CAMEL-15822: [camel-file] fileExist=Move doesn't use FileMoveExisting…

### PR DESCRIPTION
**Pull request by luigidemasi**
…Strategy when tempFile is also configured

- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: components/camel-file/src/main/java/org/apache/camel/component/file/GenericFileDefaultMoveExistingFileStrategy.java [hunk 1]
```diff
@@ -49,14 +49,24 @@ public boolean moveExistingFile(GenericFileEndpoint endpoint, GenericFileOperati
...
-        // we must normalize it (to avoid having both \ and / in the name which
-        // confuses java.io.File)
-        to = FileUtil.normalizePath(to);
+
         if (ObjectHelper.isEmpty(to)) {
             throw new GenericFileOperationFailedException(
                     "moveExisting evaluated as empty String, cannot move existing file: " + fileName);
         }
 
+        if (to.length() > 1 && to.endsWith("/")) {
```
Comment (omarsmak) [inline components/camel-file/src/main/java/org/apache/camel/component/file/GenericFileDefaultMoveExistingFileStrategy.java:58]: I see some of this logic to check the name being repeated across the three strategies, can we somehow unify this logic into a common function to ease maintainability?

Comment (luigidemasi) [reply components/camel-file/src/main/java/org/apache/camel/component/file/GenericFileDefaultMoveExistingFileStrategy.java:58]: the problem is where to put the common logic, FileUtils doesn't seem a good fit.

Comment (omarsmak) [reply components/camel-file/src/main/java/org/apache/camel/component/file/GenericFileDefaultMoveExistingFileStrategy.java:58]: By creating a singleton utility class in the component perhaps?

**Review by davsclaus — APPROVED** [review]
I am okay with the little duplicate code. However it would be good with a bit of code comments (eg // xxxx) what the purpose is for the code, as it does a bit of path manipulation but it may not be so easy to know why (and git blame may get harder to track down over time)

**Comment by luigidemasi** [comment]
> I am okay with the little duplicate code. However it would be good with a bit of code comments (eg // xxxx) what the purpose is for the code, as it does a bit of path manipulation but it may not be so easy to know why (and git blame may get harder to track down over time)

I did both, I moved the duplicated code in a common utility class and added the javadoc, hope it make sense.

**Comment by davsclaus** [comment]
Yeah this is good

**Pull request merged**